launchable/test_runners/pytest.py

```diff
@@ -62,16 +62,26 @@ def _add_testpaths(lines: List[str]):
 
 def _parse_pytest_nodeid(nodeid: str) -> TestPath:
     data = nodeid.split("::")
-    class_name = _path_to_class_name(data[0])
-    if len(data) == 3:
-        # tests/test_mod.py::TestClass::test_can_print_aaa -> tests.test_mod.TestClass
-        class_name += "." + data[1]
+    file = data[0]
+    class_name = _path_to_class_name(file)
+    
+    # file name only
+    if len(data) == 1:
+        return [
+            {"type": "file", "name": os.path.normpath(data[0])},
+            {"type": "class", "name": class_name},
+        ]
+    # file + testcase, or file + class + testcase
+    else:
+        testcase = data[-1]
+        if len(data)==3:
+            class_name += "." + data[1]
 
-    return [
-        {"type": "file", "name": os.path.normpath(data[0])},
-        {"type": "class", "name": class_name},
-        {"type": "testcase", "name": data[-1]},
-    ]
+        return [
+            {"type": "file", "name": os.path.normpath(data[0])},
+            {"type": "class", "name": class_name},
+            {"type": "testcase", "name": testcase},
+        ]
```

Comment:
I think the file variable is already defined so we should use it below to avoid confuse.

```diff
 def _parse_pytest_nodeid(nodeid: str) -> TestPath:
     data = nodeid.split("::")
-    file = data[0]
-    class_name = _path_to_class_name(file)
-
+    file = os.path.normpath(data[0])
+    class_name = _path_to_class_name(data[0])
+
     # file name only
     if len(data) == 1:
         return [
-            {"type": "file", "name": os.path.normpath(data[0])},
+            {"type": "file", "name": file},
             {"type": "class", "name": class_name},
         ]
     # file + testcase, or file + class + testcase
     else:
         testcase = data[-1]
-        if len(data)==3:
+        if len(data) == 3:
             class_name += "." + data[1]

         return [
-            {"type": "file", "name": os.path.normpath(data[0])},
+            {"type": "file", "name": file},
             {"type": "class", "name": class_name},
             {"type": "testcase", "name": testcase},
         ]
```